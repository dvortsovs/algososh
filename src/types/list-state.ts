import {ElementStates} from "./element-states";

export interface IListState<T> {
    item: T;
    state: ElementStates;
    addProgress: boolean;
    deleteProgress: boolean;
    tempItem: T;
}