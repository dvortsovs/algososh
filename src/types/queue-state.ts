import {ElementStates} from "./element-states";

export interface IQueueState<T> {
    item: (T | null)
    state: ElementStates
}