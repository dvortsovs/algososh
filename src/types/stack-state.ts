import {ElementStates} from "./element-states";

export interface IStackState<T> {
    item: T,
    state: ElementStates
}