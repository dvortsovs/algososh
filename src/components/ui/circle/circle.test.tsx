import React from 'react';
import renderer from 'react-test-renderer';
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";


describe('test circle', () => {
    test('without char', () => {
        const tree = renderer.create(<Circle />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('with char', () => {
        const tree = renderer.create(<Circle letter='i' />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('with head', () => {
        const tree = renderer.create(<Circle head='test' />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('with React element in head', () => {
        const tree = renderer.create(<Circle head={<p>test</p>} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('with tail', () => {
        const tree = renderer.create(<Circle tail='test' />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('with React element in tail', () => {
        const tree = renderer.create(<Circle tail={<p>test</p>} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('with index', () => {
        const tree = renderer.create(<Circle index={123} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('with isSmall prop', () => {
        const tree = renderer.create(<Circle isSmall />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('default state', () => {
        const tree = renderer.create(<Circle state={ElementStates.Default} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('changing state', () => {
        const tree = renderer.create(<Circle state={ElementStates.Changing} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('modified state', () => {
        const tree = renderer.create(<Circle state={ElementStates.Modified} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})