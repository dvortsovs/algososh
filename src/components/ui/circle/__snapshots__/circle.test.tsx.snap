// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test circle changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle with React element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <p>
      test
    </p>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle with React element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <p>
      test
    </p>
  </div>
</div>
`;

exports[`test circle with char 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      i
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    123
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`test circle with isSmall prop 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test
  </div>
</div>
`;

exports[`test circle without char 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
